### PR TITLE
Save favorites query string to GC datastore

### PR DESCRIPTION
**Pull request by lovedeep92**
**JIRA:**
See ticket here - https://jira.rax.io/browse/MNRVA-823

**Description:**

1) Add logic for storing query api url to GCP datastore.
2) Create favorites service.
3) call minerva api to avoid CORS issue.

 **Testing:**

`npm run test:unit`

 **Screenshots:**

Screenshot 2021-08-03 at 11 33 02 AM

(if applicable)

### CODE REVIEW

**Review by FarajiA — APPROVED** [review]
Cool cool.

File: package.json [hunk 1]
```diff
@@ -58,7 +58,7 @@
     "ajv": "6.12.0",
     "core-js": "2.5.4",
     "firebase": "8.1.2",
-    "hedwig-monitoring-library": "https://github.com/racker/Hedwig/releases/download/v1.3.0/hedwig-monitoring-library.tgz",
+    "hedwig-monitoring-library": "https://github.com/racker/Hedwig/releases/download/v1.1.2/hedwig-monitoring-library.tgz",
```
Comment (pras7430) [inline package.json:61]: this is old version

File: src/app/_shared/components/hedwig-graph/hedwig-graph.component.ts [hunk 2]
```diff
@@ -46,9 +49,38 @@ export class HedwigGraphComponent implements OnInit {
       )
   }
 
-
-  setFav() {
+  setFav(metricName) {
```
Comment (pras7430) [inline src/app/_shared/components/hedwig-graph/hedwig-graph.component.ts:52]: this should be  at parent component

File: src/app/_shared/components/hedwig-graph/hedwig-graph.component.ts [hunk 3]
```diff
@@ -26,13 +28,14 @@ export class HedwigGraphComponent implements OnInit {
   @Input() group: string;
   @Input() unit:string="percentage";
 
+  
 
   lineGraph: LineGraph = LineGraph;
 
-  constructor() { }
+  constructor( private favoritesService: FavoritesService ) {  }
```
Comment (pras7430) [inline src/app/_shared/components/hedwig-graph/hedwig-graph.component.ts:35]: Hedwig graph component should not deal with service directly.

**Review by pras7430 — COMMENTED** [review]
As graph component purpose is to show graphs not to save in any database, so we should not use any service at graphcomonent dirtectly

File: src/app/_shared/components/hedwig-graph/hedwig-graph.component.ts (hunk 3 above)
Comment (lovedeep92) [reply src/app/_shared/components/hedwig-graph/hedwig-graph.component.ts:35]: okay .. done!

File: package.json (hunk 1 above)
Comment (lovedeep92) [reply package.json:61]: fixed!

File: src/app/_features/visualize/pages/visualize.page/visualize.page.component.ts [hunk 4]
```diff
@@ -317,6 +318,36 @@ export class VisualizePage {
     
 }
 
+  countChangedHandler(event:any) {
+      let url = location.href;
```
[the remark this answers is not in the document]
Comment (lovedeep92) [reply src/app/_features/visualize/pages/visualize.page/visualize.page.component.ts:322]: got it .. I am using the properties which is set inside set query params function!

File: src/app/_features/visualize/pages/visualize.page/visualize.page.component.ts [hunk 5]
```diff
@@ -317,6 +318,36 @@ export class VisualizePage {
...
+      if(hash==-1 && questionMark==-1) return {};
+      if(hash==-1) hash = url.length;
+      var query = questionMark==-1 || hash==questionMark+1 ? url.substring(hash) : 
+      url.substring(questionMark+1,hash);
+      var part = query.split("&");
+
+       var startTime = part[1].split("=")[1];
+       var tagValue = part[2].split("=")[1];
+          
+      var data = {
```
[the remark this answers is not in the document]
Comment (lovedeep92) [reply src/app/_features/visualize/pages/visualize.page/visualize.page.component.ts:334]: I am directly using tagPillSet and date as It always set whenever url loaded!

**Pull request merged**